integration-app/src/main/java/uk/gov/crowncommercial/esourcing/integration/service/TenderApiService.java

```diff
@@ -161,38 +172,88 @@ public TenderApiService(
   }
 
   @Override
-  public ResponseEntity<AbandonITT200Response> invalidITT(AbandonITT abandonITT){
-
-    AbandonITT200Response response = new AbandonITT200Response();
+  public ResponseEntity<String> terminateRFx(String procID, String eventID) {
+
+    logger.info("Invalidating ITT  - ProcID : {} / EventID : {}", procID, eventID);
+
+    RfxGetResponse rfx =
+        rfxApi
+            .getRFX(String.format("rfxReferenceCode==%s", eventID))
+            .block(Duration.ofSeconds(defaultApiTimeout));
+
+    final RfxSetting rfxSetting;
+    if (rfx != null) {
+      if (rfx.getDataList() != null) {
+        if (rfx.getDataList().getRfx() != null) {
+          rfxSetting =
+              rfx.getDataList() != null ? rfx.getDataList().getRfx().get(0).getRfxSetting() : null;
```

Comment:
Lots of checks making sure data is correct.
+ Have already checked `(rfx.getDataList() != null)`
+ Missing one that checks the list has at least one element before getting the first element.

This is a bit arrow anti-pattern like. I prefer 

```
if (rfx != null) {
   throw new IllegalStateException("...")
}
if (rfx.getDataList != null) {
   throw new IllegalStateException("...")
}
```

Reply:
Good point - problem with fixing null pointer warnings a different times